get_http_test.go

```diff
@@ -181,28 +187,39 @@ func TestHttpGetter_resume(t *testing.T) {
 	if err != nil {
 		t.Fatalf("create: %v", err)
 	}
-	f.Write(load[:downloadFrom])
-	f.Close()
+	if n, err := f.Write(load[:downloadFrom]); n != downloadFrom || err != nil {
+		t.Fatalf("partia file write failed: %d, %s", n, err)
+	}
+	if err := f.Close(); err != nil {
+		t.Fatalf("close failed: %s", err)
+	}
 
-	var u url.URL
-	u.Scheme = "http"
-	u.Host = ln.Addr().String()
-	u.Path = "/range"
+	u := url.URL{
+		Scheme:   "http",
+		Host:     ln.Addr().String(),
+		Path:     "/range",
+		RawQuery: "checksum=" + checksum,
+	}
+	t.Logf("url: %s", u.String())
 
-	// Get it!
-	if err := g.GetFile(dst, &u); err != nil {
-		t.Fatalf("should not error: %v", err)
+	// Finish getting it!
+	if err := GetFile(dst, u.String()); err != nil {
+		t.Errorf("finishing download should not error: %v", err)
```

Comment:
Is the test worth continuing here if this fails?

Reply:
In my case it was, because the checksum failed here, but then the next checksum passed. ( I just had to `hash.Reset()` to fix )

Reply:
This gave me insightful information as to where something was wrong, but may be now that this issue is fixed, do you think I should rather `Fatal` ?

Reply:
I think that maybe the spirit of `Errorf` is useful when you need multiple assertions in a single test.

Consider:

```go
func TestFoo(t *testing.T) {
  x, err := foo("a", "b")
  if err != nil {
    t.Fatalf("error: %s", err)
  }

  if x.A != "a" {
    t.Errorf(`x.A: expected "a", got %q`, x.A)
  }

  if x.B != "B" {
    t.Errorf(`x.B: expected "b", got %q`, x.B)
  }
}
```

In this application, `Errorf` allows you to check both fields in the same test and display failures if _both_ of them fail at the same time, versus having field `A` take priority. However, if there was an error funning `foo`, then more than likely the result in `x` will be invalid and not suitable for giving valid test results (fields could be zero values, or `x` could be `nil`, causing a panic).

In this application, you are not asserting multiple final results, you are mainly testing for a final result of success (no failures). Each failure down the test case ultimately renders the rest of the function unstable, or it be should considered as such. Hence, while for debugging it might have been useful, from a testing standpoint strictly speaking it's not, as an early failure is going to affect everything below it, possibly giving incorrect and/or non-useful error data for operations afterwards.

Reply:
Ah, yes, 👍 awesome explanation, I'll make them Fatal 🙂

Reply:
Thanks !!